d2-schemas/src/main/pegasus/com/linkedin/d2/D2Service.pdl

```diff
@@ -96,4 +96,9 @@ record D2Service includes D2ChangeTimeStamps {
    * Configuration of backup requests. Each element describes backup requests configuration for specific operation.
    */
   backupRequests: optional array[BackupRequestsConfiguration]
+
+  /**
+   * The minimum cluster subset size for this service. With a value greater than 0, each client will only send requests to a subset of the hosts in the cluster. A value of -1 disables the feature.
+   */
+  minClusterSubsetSize: optional int
```

Comment:
Another question about this subsetSize - is this size the number of physical host size or weight size? For example, if a host has the weight of 20.0, in the same cluster there are 10 other hosts with weight 1.0, which means that redlined host should get 2/3 of the traffic. In this way, if the subset size is 5 for example, what does 5 mean here?

Reply:
Strictly speaking, minClusterSubsetSize of _s_ is equivalent to a fixed aperture width that contains _s_ hosts **when all hosts have weight 1.0**. But since this definition contains too many terminology, and d2 weight is 1.0 most of the time, I simplify the description a little bit.

For your example, the minimum subset will occupy (5/30 = 1/6) of the ring. When the host weights change, there may be more/less than 5 hosts in the subset.

Reply:
That makes sense. Thank you